src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.cs

```diff
@@ -58,47 +59,79 @@ private void EnsureWriteCapacity(int pendingCount)
             }
         }
 
-        private void EnsureCanWriteNewDataItem()
-        {
-            if (_remainingDataItems == 0)
-            {
-                throw new InvalidOperationException("Adding a CBOR data item to the current context exceeds its definite length.");
-            }
-        }
-
         private void PushDataItem(CborMajorType type, uint? expectedNestedItems)
         {
-            _nestedDataItemStack ??= new Stack<(CborMajorType, uint?)>();
-            _nestedDataItemStack.Push((type, _remainingDataItems));
+            _nestedDataItemStack ??= new Stack<(CborMajorType, bool, uint?)>();
```

Comment:
This nesting stack tuple is getting pretty close to its comprehendability limit (if it didn't already pass it).  If it grows any more it warrants a named struct type to help control line length and provide clarity.

Reply:
I'm using named elements in the field declaration, only omitted here for brevity. I'd argue that 3 elements are probably ok, would need to revisit though if new ones need to be added.